lib/web/device_trust.go

```diff
@@ -76,7 +81,29 @@ func (h *Handler) deviceWebConfirm(w http.ResponseWriter, r *http.Request, _ htt
 
 	// Always redirect back to the dashboard, regardless of outcome.
 	app.SetRedirectPageHeaders(w.Header(), "" /* nonce */)
-	http.Redirect(w, r, "/web", http.StatusSeeOther)
+
+	redirectTo := "/web"
+
+	if redirectParam != "" {
+		parsedURL, err := url.Parse(redirectParam)
+		if err == nil {
+			cleanPath := path.Clean(parsedURL.Path)
+			// helps in situations where there is no path such as https://example.com
+			if cleanPath == "." || cleanPath == "" {
```

Comment:
Do we need to handle ".." too?

Reply:
`Clean` will only return the path, "/" or ".", with all ".." removed so I don't believe we need the check here. I can also remove the empty string check, thank you!

Reply:
path.Clean("..") does return "..", so there is that to consider.